ui/pages/confirmations/hooks/alerts/useDomainMismatchAlert.ts

```diff
@@ -0,0 +1,46 @@
+import { useMemo } from 'react';
+import {
+  isValidSIWEOrigin,
+  WrappedSIWERequest
+} from '@metamask/controller-utils';
+
+import { Alert } from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { SignatureRequestType } from '../../types/confirm';
+import { isSIWESignatureRequest } from '../../utils';
+import useCurrentConfirmation from '../useCurrentConfirmation';
+
+const useDomainMismatchAlerts = (): Alert[] => {
+  const { currentConfirmation } = useCurrentConfirmation();
```

Comment:
When querying the confirmation, we want to use the Redux cached state, rather than deriving it every time, so should we use `currentConfirmationSelector`?

Reply:
I agree. Updated 8bb125bab54859f2178d2182d3a5e1036918add2

I also think we'll want this in useBlockaidAlert. Updated c21feb210598157a5ca1b5c667ebb5324dfb1ab0